### PR TITLE
Select/Disable plugins via command line args

### PR DESCRIPTION
**Pull request by Atlante45**
This PR adds the ability to disable (not even load) plugins via command line argument.
As well as specify an ordered list of preferred display plugins.

Examples:
`./interface.exe --display "Oculus Rift"`
This will start with the oculus display plugin if available or fallback to the settings

`./interface.exe --display "Oculus Rift","OpenVR (Vive)"`
This will try to start with the oculus display plugin, then the Vive display plugin then fallback to the settings if all are unavailable.

`./interface.exe --disable-displays "Oculus Rift","OpenVR (Vive)" --disable-inputs "Sixense","OpenVR (Vive)","Oculus Rift","SDL2"`
This will prevent the following plugins from loading:
- Oculus Rift display plugin
- OpenVR display plugin
- Sixense input plugin
- Oculus Rift input plugin
- OpenVR input plugin

### CODE REVIEW

**Comment by Atlante45** [comment]
Test plan:
- Verify that normally using interface saves/loads last display plugin selected.
- Running via command line, make sure you can select a different plugin that the one stored int the settings
- Running via command line, make sure when disabling plugins, they do not show up in the menu

**Comment by zzmp** [comment]
Can you document this somewhere?

**Comment by zzmp** [comment]
Is it possible to refer to things by multiple names? It seems more user-friendly to be able to call the legacy oculus `DK1` or `DK2`, and the OpenVR `Vive`, without all of the meta around it.

**Comment by Atlante45** [comment]
@zzmp Not sure what you mean. You can select via the name of the plugin.
DK1(?), DK2 and CV1 use the same plugin on Win: "Oculus Rift"
If you want the DK2 on Mac, then that would be "Oculus Rift (0.5) (Legacy)"

**Comment by zzmp** [comment]
I mean that "Oculus Rift (0.5) (Legacy)" seems very verbose, but this seems like a dev feature anyways, so it's ok.

**Comment by ZappoMan** [comment]
How about we rename it "Oculus DK2" since it's only available on the Mac
and it only currently supports DK2?

**Comment by hifi-gustavo** [comment]
Build console output for commit: a92f4249c59191aff76418e3707b758a0c6dfcf6 available at: https://gist.github.com/c07f5e4c2d03ca5c37ceb20097f07759<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.dmg

**Comment by zzmp** [comment]
I don't think "Oculus DK2" is good because the DK2 works with the newer runtime. Something mac-specific would be OK, though, like "Oculus (Mac)". As is is also fine by me.

**Comment by Atlante45** [comment]
In that case I think the name should be "Oculus Rift (Legacy)" because technically it could run on windows I'm pretty sure, we only compile it for mac because it's not needed on windows.
But in the end "Oculus Rift (0.5) (Legacy)" describes perfectly what the plugin is, I'm not convinced it needs changing.
It's not like you're gonna have to type(copy/paste?) it every day.

**Comment by ZappoMan** [comment]
Gak - c'mon... we only compile it on Mac... no one is going to run it on windows... Let's just call it "Oculus Rift"... we don't have to worry about legacy vs new confusion

**Comment by Atlante45** [comment]
Sold

**Comment by hifi-gustavo** [comment]
Build console output for commit: a627abc393edd86c9c7c63fa90ef5576a0ed48b8 available at: https://gist.github.com/c07f5e4c2d03ca5c37ceb20097f07759<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f269890c69878385efc0a147dc0041443cc9791b available at: https://gist.github.com/c07f5e4c2d03ca5c37ceb20097f07759<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.dmg

File: interface/src/Application.cpp [hunk 1]
```diff
@@ -2943,6 +2943,19 @@ void Application::loadSettings() {
...
     Menu::getInstance()->loadSettings();
+
+    // If there is a preferred plugin, we probably messed it up with the menu settings, so fix it.
+    auto plugins = PluginManager::getInstance()->getPreferredDisplayPlugins();
+    for (auto plugin : plugins) {
+        auto menu = Menu::getInstance();
+        if (auto action = menu->getActionForOption(plugin->getName())) {
+            action->setChecked(true);
+            action->trigger();
+            // Find and activat5ed highest priority plugin, bail for the rest
```
Comment (jherico) [inline interface/src/Application.cpp]: typo in activate

**Comment by hifi-gustavo** [comment]
Build console output for commit: 6bd70d5d62a3cb0c1e4a84b1def23c093556ed64 available at: https://gist.github.com/c07f5e4c2d03ca5c37ceb20097f07759<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: f701859837d5757dbcc8c785492425e787525ad6 available at: https://gist.github.com/c07f5e4c2d03ca5c37ceb20097f07759<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.dmg

**Comment by cozza13** [comment]
I crash when i launch. (Just clicking on the interface icon) bugsplat submitted.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 399517fcbd54ab7a6c6825bd9360528421b32023 available at: https://gist.github.com/c07f5e4c2d03ca5c37ceb20097f07759<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR8000.dmg

**Comment by cozza13** [comment]
<a href="https:&#x2F;&#x2F;trello.com&#x2F;c&#x2F;UJsNLLQe&#x2F;10-allow-interface-to-launch-to-vive-via-command-line"> Allow Interface to launch to Vive via command line</a>

**Pull request merged**